src/JustSaying/AwsTools/MessageHandling/SqsQueueByNameBase.cs

```diff
@@ -1,25 +1,45 @@
 using System;
 using System.Collections.Generic;
+using System.Net;
 using System.Threading.Tasks;
 using Amazon;
 using Amazon.SQS;
 using Amazon.SQS.Model;
 using JustSaying.AwsTools.QueueCreation;
 using JustSaying.Extensions;
+using JustSaying.Messaging.Interrogation;
 using Microsoft.Extensions.Logging;
 
 namespace JustSaying.AwsTools.MessageHandling
 {
     [Obsolete("SqsQueueBase and related classes are not intended for general usage and may be removed in a future major release")]
-    public abstract class SqsQueueByNameBase : SqsQueueBase
+    public abstract class SqsQueueByNameBase : ISqsQueue
     {
         protected SqsQueueByNameBase(RegionEndpoint region, string queueName, IAmazonSQS client, ILoggerFactory loggerFactory)
-            : base(region, client, loggerFactory)
         {
+            if (loggerFactory == null) throw new ArgumentNullException(nameof(loggerFactory));
             QueueName = queueName;
+            Client = client;
+            Region = region;
+            Logger = loggerFactory.CreateLogger("JustSaying");
         }
 
-        public override async Task<bool> ExistsAsync()
+        public string Arn { get; protected set; }
+        public Uri Uri { get; protected set; }
+        public IAmazonSQS Client { get; private set; }
+        public string QueueName { get; protected set; }
+        public RegionEndpoint Region { get; protected set; }
+        public ErrorQueue ErrorQueue { get; protected set; }
+        internal TimeSpan MessageRetentionPeriod { get; set; }
+        internal TimeSpan VisibilityTimeout { get; set; }
+        internal TimeSpan DeliveryDelay { get; set; }
+        internal RedrivePolicy RedrivePolicy { get; set; }
+        internal ServerSideEncryption ServerSideEncryption { get; set; }
+        public string Policy { get; private set; }
+        public string RegionSystemName { get; }
+        protected ILogger Logger { get; }
+
+        public virtual async Task<bool> ExistsAsync()
```

Comment:
Should we add `CancellationToken` parameters to these various methods?

Reply:
We should, I'll hold off for a separate PR though, this ones getting a little heavy change wise